Create hercules_djcontrol_jogvision.rst (#134)
mixxxdj/manual: source/hardware/controllers/hercules_djcontrol_jogvision.rst

Hercules DJControl Jogvision
============================

-  `Manufacturer's product page <https://www.hercules.com/en-us/product/djcontroljogvision-2old/>`__
-  `Manufacturer's support and downloads page <https://support.hercules.com/en/product/djcontroljogvision-en/>`__
-  `Forum thread <https://www.mixxx.org/forums/viewtopic.php?f=7&t=12580>`__
  
Whether you’re just getting started DJing or you’ve already refined your skills, you can let your creativity run wild.
Enjoy excellent precision and comfort to create your mixes using the circular displays on the jog wheels to guide your movements and perfect your scratches,
and have fun with the amazing contactless AIR Control sensor.
Your adventure starts right here and now. It’s your turn\!

Compatibility
-------------

This controller is a class compliant USB MIDI and audio device, so it
can be used without any special drivers on GNU/Linux, Mac OS X, and
Windows. For Windows, please install the latest driver package available from
the `Support
page <https://support.hercules.com/es/product/djcontroljogvision-es/>`__.

Sound card setup
----------------

This controller has built-in 4 channel output sound card, with:

> - Outputs:
> 
>   - MASTER: RCA and 3.5mm jack
>   - HEADPHONE: 6.35mm and 3.5mm jack
>   - BOOTH: RCA and 3.5mm jack
> - Inputs:
> 
>   - MIC: 6.35mm jack
>   - AUX: 3.5mm jack

Reviewer: Maybe a table would look nicer?

```suggestion
================  ===================
Channel           Port
================  ===================
Master Output     RCA and 3.5mm jack
Booth Output      RCA and 3.5mm jack
Headphone Output  6.35mm and 3.5mm jack
Microphone Input  6.35mm jack
Auxiliary Input   3.5mm jack
================  ===================
```

> - Open **Preferences \> Sound Hardware**  
> - Select the **Output** tab  
> - From the **Master** drop-down menu, select the audio interface, then **Channels 1-2**  
> - From the **Headphones** drop-down menu, select the audio interface, then **Channels 3-4**  
> - Click **Apply** to save the changes.  

Reviewer: ```suggestion
- Open :menuselection:`Preferences --> Sound Hardware`  
- Select the :guilabel:`Output` tab  
- From the :guilabel:`Master` drop-down menu, select the audio interface, then :guilabel:`Channels 1-2` 
- From the :guilabel:`Headphones` drop-down menu, select the audio interface, then :guilabel:`Channels 3-4`  
- Click :guilabel:`Apply` to save the changes.  
```

Please refer to the user manual for more details about the audio configuration in Mixxx.

.. note::

>    The **Master** and **Headphone** knobs, as well as the
>    **Master** button are hardware controls and interact directly with the

Reviewer: ```suggestion
   The :hwlabel:`MASTER` and :hwlabel:`HEADPHONE` knobs, as well as the
   :hwlabel:`MASTER` button are hardware controls and interact directly with the
```

.. list-table:: *FINE TUNNING Variables* (you may modify them, but not too far from the default values...)
   :widths: 25 25 25 50
   :header-rows: 1

   * - VARIABLE NAME
     - DEFAULT VALUE
     - AVAILABLE VALUES
     - DESCRIPTION
   * - beatHelpSensitivity
     - 0.5
     - \[0..1)
     - Max distance (float) in BPMs to be considered as a match (in use if CFG.user.beatHelper=1): the lower, the more sensitive. Values equal or bigger than 1 are reset to 0.9 and a warning is printed
   * - quickMoveFactor
     - 0.002
     - \[0..1\]
     - the smaller (float), the slower MODE+JogWheel will move 'playposition' (when such channel is NOT playing)
   * - quickBrowseFactor
     - 10
     - \[0..inf\]
     - the bigger, the faster MODE+Browser jog will move the cursor position in the library up/down
   * - spinBackBrakeFactor
     - 100
     - (0..5000\]
     - the bigger, the softer the brake will be applied (0 = immediate stop; >=5000 = it will take almost forever to stop)
   * - spinBackInitialSpeed
     - 6
     - (0..200\]
     - the bigger, the stronger will be the "back" impulse (1 = no spinbak, but stop and start sloooowly)
   * - mixGainFactor
     - 0.1
     - (0..1)

>      - (float) the bigger, the faster the Pregain or Mix level will be updated

Reviewer: This table syntax is weird and very hard to edit. Also, the table expands too wide.

I'd suggest using a definition list (see below), but tbh I think it makes more sense to document these user options in the source file itself rather than in the manual (where it could confuse users). Let's discuss this on Zulip: https://mixxx.zulipchat.com/#narrow/stream/109176-documentation/topic/Built-in.20controller.20mapping.20documentation.20in.20the.20manual/near/213411520

```rest
User Variables
++++++++++++++

These variables can be modified freely.

beatDetection
  Either set to ``normal`` or ``follow`` (default is ``normal``).
  When set to ``follow``, the song position is more accurate with the beat position, but beat detection is much less accurate.

initUpdateEffects
   Sets some effect values at startup with the given floating-point value (between 0.0 and 1.0, default is 0.8).

...
```

Reviewer: On second thought, please replace this section with a simple hint and move all the documention into the JavaScript file itself:

```suggestion
.. hint::
   This controller mapping can be customized by editing the corresponding Javascript file and editing the configuration options at the top of the file.
```

> All controls *act as labeled*, but there are some **additional** functionalities that have been added:

Reviewer: ```suggestion
All controls act as labeled, but there are some additional functionalities that have been added:
```

> - MODE+Loop ON                  : set a loop_in mark (with curently defined loop_size), activate it, and enable slip mode  

Reviewer: Can you please run `git pull https://github.com/mixxxdj/manual.git manual-2.3.x`, then use the `:hwlabel:` role for button/knob/fader labels on the controller?

Like this:
```suggestion
- :hwlabel:`MODE` + :hwlabel:`LOOP ON`                  : set a loop_in mark (with curently defined loop_size), activate it, and enable slip mode  
```

> - MODE+Loop ON                  : set a loop_in mark (with curently defined loop_size), activate it, and enable slip mode  
> - MODE+Loop X 1/2 / X 2         : do a 'beatjump_size' beats beatjump backward/forward  
> - MODE+Loop Size Knob           : decrease/increase pitch (only key, not tempo!)  
> - MODE+JogWheel plate (playing) : scratch with 'Slip' ON (deactivate 'Slip' when plate is released)  
> - MODE+JogWheel plate (stopped) : move song position backward/forward faster by 'quickMoveFactor' factor  
> - MODE+Browser Knob Turn        : move library selected position in groups of 'quickBrowseFactor' elements forward/backward  
> - MODE+LOAD A|B                 : toggle 'quantize' for deck where MODE key is pressed  
> - SHIFT+LOAD A|B                : eject track from deck where SHIFT key is pressed  
> - SHIFT+Browser Knob Press      : activate (double-click) currently selected item in browser  
> - SHIFT+Loop Size Knob          : move existing loop forward/backward  
> - SHIFT+JogWheelTouch           : do a 'backspin' with 'spinBackBrakeFactor' and 'spinBackInitialSpeed' factors  
> - SHIFT+MultiFX                 : set beatgrid to current position  
> - SHIFT+Air control Filter      : do the reverse than standard, that is, high-pass filter  

Reviewer: Please run `git pull https://github.com/mixxxdj/manual.git manual-2.3.x` and push to this branch, then apply this suggestion:

```suggestion
=========================================================  =============================================================
Control                                                    Description
=========================================================  =============================================================
:hwlabel:`MODE` + :hwlabel:`LOOP ON                        Set a loop_in mark (with curently defined loop_size), activate it, and enable slip mode
:hwlabel:`MODE` + :hwlabel:`LOOP X½` / :hwlabel:`LOOP X2`  Do a beatjump backward/forward
:hwlabel:`MODE` + :hwlabel:`LOOP SIZE` knob                Decrease/Increase pitch (only key, not tempo!)
:hwlabel:`MODE` + Jog Wheel plate (playing)                Scratch with 'Slip' on (deactivate 'Slip' when plate is released)
:hwlabel:`MODE` + Jog Wheel plate (stopped)                Move song position backward/forward faster
:hwlabel:`MODE` + :hwlabel:`BROWSER` knob turn             Move library selected position multiple elements forward/backward
:hwlabel:`MODE` + :hwlabel:`LOAD` buttons                  Toggle 'quantize' for deck where :hwlabel:`MODE` button is pressed
:hwlabel:`SHIFT` + :hwlabel:`LOAD` buttons                 Eject track from deck where :hwlabel:`SHIFT` key is pressed
:hwlabel:`SHIFT` + :hwlabel:`BROWSER` knob press           Activate (double-click) currently selected item in browser
:hwlabel:`SHIFT` + :hwlabel:`LOOP SIZE` knob               Move existing loop forward/backward
:hwlabel:`SHIFT` + Jog Wheel touch                         Do a backspin
:hwlabel:`SHIFT` + `:hwlabel:`MULTI FX`                    Set beatgrid to current position
:hwlabel:`SHIFT` + :hwlabel:`AIR CONTROL` Filter           Do the reverse than standard, that is, high-pass filter
=========================================================  =============================================================
```

Hercules DJControl Jogvision
============================

-  `Manufacturer's product page <https://www.hercules.com/en-us/product/djcontroljogvision-2old/>`__
-  `Manufacturer's support and downloads page <https://support.hercules.com/en/product/djcontroljogvision-en/>`__
-  `Forum thread <https://www.mixxx.org/forums/viewtopic.php?f=7&t=12580>`__
  
Whether you’re just getting started DJing or you’ve already refined your skills, you can let your creativity run wild.
Enjoy excellent precision and comfort to create your mixes using the circular displays on the jog wheels to guide your movements and perfect your scratches,
and have fun with the amazing contactless AIR Control sensor.
Your adventure starts right here and now. It’s your turn\!

Compatibility
-------------

This controller is a class compliant USB MIDI and audio device, so it
can be used without any special drivers on GNU/Linux, Mac OS X, and
Windows. However, if you wish to use the `ASIO sound

> API <https://mixxx.org/manual/latest/en/chapters/preferences.html?highlight=asio#windows>`__

Reviewer: Please replace all links to the manual with internal cross references:

    :ref:`Some text <some-cross-reference>`.

The cross-references are defined in the respective file *above* the headline you want to link to like this: `.. _some-coss-reference:`. See this for details:  https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html

Otherwise these links won't work as intended when generating a non-web manual (PDF, Qt Help, etc.).

under Windows, please install the latest driver package available from
the `Support
page <https://support.hercules.com/en/product/djcontrolinpulse300-en//>`__.

Sound card setup
----------------

This controller has built-in 4 channel output sound card, with:

> - OUTPUTS:

Reviewer: ```suggestion
- Outputs:
```

>   - MASTER:RCA and 3.5mm jack

Reviewer: ```suggestion
  - MASTER: RCA and 3.5mm jack
```

  - HEADPHONE: 6.35mm and 3.5mm jack
  - BOOTH: RCA and 3.5mm jack

> - INPUTS:

Reviewer: ```suggestion
- Inputs:
```

>   - MIC:6.35mm jack

Reviewer: ```suggestion
  - MIC: 6.35mm jack
```

  - AUX: 3.5mm jack

- Open **Preferences \> Sound Hardware**  
- Select the **Output** tab  
- From the **Master** drop-down menu, select the audio interface, then **Channels 1-2**  
- From the **Headphones** drop-down menu, select the audio interface, then **Channels 3-4**  
- Click **Apply** to save the changes.  

> Please refer to `the user manual <https://mixxx.org/manual/latest/en/chapters/example_setups.html#laptop-and-external-usb-audio-interface>`__

Reviewer: Use cross references here (see above)

for more details about the audio configuration in Mixxx.

.. note::
   The **Master** and **Headphone** knobs, as well as the
   **Master** button are hardware controls and interact directly with the
   integrated sound card’s output.
   
Mapping description
-------------------

> Save both MIDI and script files to your `user controller mapping

Reviewer: I don't think this paragraph is needed and could confuse users. We'll only merge this when the mapping PR has been merged as well. In that case, the mapping is built-in and it suffices to select the correct mapping from the drop down menu.

folder <https://github.com/mixxxdj/mixxx/wiki/controller%20mapping%20file%20locations#user%20controller%20mapping%20folder>`__,
then load the preset as described in `the user
manual <https://mixxx.org/manual/latest/en/chapters/controlling_mixxx.html#using-midi-hid-controllers>`__

Mapping configuration
~~~~~~~~~~~~~~~~~~~~~

> Edit the 'Hercules_DJControl_Jogvision-scripts.js', and change the 'CFG.(user|fine)' variables to fit your needs:  

Reviewer: ```suggestion
Edit the :file:`Hercules_DJControl_Jogvision-scripts.js`, and change the ``CFG.(user|fine)`` variables to fit your needs:  
```

- From the :guilabel:`Headphones` drop-down menu, select the audio interface, then :guilabel:`Channels 3-4`  
- Click :guilabel:`Apply` to save the changes.  

Please refer to the user manual for more details about the audio configuration in Mixxx.

.. note::
   The :hwlabel:`MASTER` and :hwlabel:`HEADPHONE` knobs, as well as the
   :hwlabel:`MASTER` button are hardware controls and interact directly with the
   integrated sound card’s output.
   
Mapping configuration
~~~~~~~~~~~~~~~~~~~~~

Edit the :file:`Hercules_DJControl_Jogvision-scripts.js`, and change the ``CFG.(user|fine)`` variables to fit your needs:  

.. hint::
   This controller mapping can be customized by editing the corresponding Javascript file and editing the configuration options at the top of the file.

All controls act as labeled, but there are some additional functionalities that have been added:

=========================================================  =============================================================
Control                                                    Description
=========================================================  =============================================================

> :hwlabel:`MODE` + :hwlabel:`LOOP ON                        Set a loop_in mark (with curently defined loop_size), activate it, and enable slip mode

Reviewer: ```suggestion
:hwlabel:`MODE` + :hwlabel:`LOOP ON`                       Set a loop_in mark (with curently defined loop_size), activate it, and enable slip mode
```
